--- b/src/media/gpu/gpu_video_decode_accelerator_factory.h	2017-06-26 16:05:55.402992430 +0900
+++ src/media/gpu/gpu_video_decode_accelerator_factory.h	2017-06-26 14:38:33.461719218 +0900
@@ -97,6 +97,11 @@
       const gpu::GpuDriverBugWorkarounds& workarounds,
       const gpu::GpuPreferences& gpu_preferences) const;
 #endif
+#if defined(IMX_PLATFORM)
+  std::unique_ptr<VideoDecodeAccelerator> CreateImxVpuVDA(
+      const gpu::GpuDriverBugWorkarounds& workarounds,
+      const gpu::GpuPreferences& gpu_preferences) const;
+#endif
 #if defined(OS_CHROMEOS) && defined(USE_V4L2_CODEC)
   std::unique_ptr<VideoDecodeAccelerator> CreateV4L2VDA(
       const gpu::GpuDriverBugWorkarounds& workarounds,
